tests/unit/metrics_test.cc

```diff
@@ -319,3 +320,57 @@ SEASTAR_THREAD_TEST_CASE(test_relabel_enable_disable_skip_when_empty) {
     }), 0);
     sm::set_relabel_configs({}).get();
 }
+
+SEASTAR_THREAD_TEST_CASE(test_estimated_histogram) {
+    using namespace seastar::metrics;
+    using namespace std::chrono_literals;
+    time_estimated_histogram histogram1;
+    time_estimated_histogram histogram2;
+    time_estimated_histogram::duration min = std::chrono::microseconds(512);
+    time_estimated_histogram::duration next = min*2;
+
+    for (size_t i = 0; i < 16; i++) {
+        auto delta = (next - min)/4;
```

Comment:
nit: magic number. It would be better to have something like:
```
// The number of linearly-spaced buckets between consecutive powers of 2 in time_estimated_histogram.
constexpr int PRECISION = 4;
...
auto delta = (next - min) / PRECISION;
```